lib/src/reactive_ble.dart

```diff
@@ -220,14 +220,22 @@ class FlutterReactiveBle {
         .then((message) => message.result.dematerialize());
   }
 
-  /// Scan for devices that are advertising a specific service
-  /// [scanMode] is used only on Android to enforce a more battery or latency intensive scan strategy
-  ///
-  Stream<DiscoveredDevice> scanForDevices(
-      {@required Uuid withService, ScanMode scanMode = ScanMode.balanced}) {
+  /// Scan for devices that are advertising a specific services. Use [withServices] filter on devices that are advertising
+  /// the defined services. 
+  /// 
+  /// There are two Android specific parameters that will be ignored on iOS: 
+  /// [scanMode] is used to enforce a more battery or latency intensive scan strategy,
+  /// [locationServiceIsMandatory] is used to enforce location services to be enabled while scanning.
+  /// It is recommended to set this to true when supporting Android 7 and lower.
```

Comment:
Does the library support Android versions lower than 7?

Reply:
I'm following this PR very closely because it would allow me to switch to flutter_reactive_ble. Is it possible to lower the minsdk version to at least 21 if there aren't any technical hurdles? It looks like other libraries based on rxandroidble supports android SDK from 18 onwards and would be good to have support for device which are a bit older but still make up for the vast majority of market share

Reply:
When we started with the ble library we decided from beginning to support 4 Android versions (7,8,9 and 10). Reason is the fragmentation on Android and the fact BLE behaves totally different on Android versions below 7. We prefer to be good on a more limited range of Android versions. Of course if there is more demand from the community and ppl are willing to help us out we can consider supporting more Android versions.